scripts/core/editor3/components/DraftSpellchecker.jsx

```diff
@@ -0,0 +1,244 @@
+        this.getWords = getWords;
+        this.getIgnoredWords = getIgnoredWords;
+        this.initWords(false);
+    }
+
+    /**
+      * @ngdoc method
+      * @name DraftSpellchecker#initWords
+      * @param {Boolean} refresh if true force the loading of the last version of lists from backend
+      * @description Get the list of words and ignored words
+      */
+    initWords(refresh) {
+        this.ignoredWords = this.getIgnoredWords();
+        this.getWords(this.language, refresh).then((words) => {
+            this.words = words;
+            this.initDecorators();
+        });
+    }
+
+    /**
+      * @ngdoc method
+      * @name DraftSpellchecker#isCorrectWord
+      * @param {String} word the current word
+      * @description Check if the word is correct
+      */
+    isCorrectWord(word) {
+        if (!this.words) {
+            return true;
+        }
+        return this.words[word] || this.ignoredWords[word];
+    }
+
+    /**
+      * @ngdoc method
+      * @name DraftSpellchecker#getDecorator
+      * @return {Object} returns the decorator
+      * @description Decorator for spellchecker, will highlight the words with errors
+      */
+    getDecorator() {
+        return {
+            strategy: this.spellcheckStrategy(this.isCorrectWord),
+            component: this.spellcheckSpan,
+            props: {showContextMenu: this.showContextMenu}
+        };
+    }
+
+    /**
+      * @ngdoc method
+      * @name DraftSpellchecker#spellcheckStrategy
+      * @return {Object} returns the strategy
+      * @description For a block check the words that has errors
+      */
+    spellcheckStrategy(isCorrectWord) {
+        let WORD_REGEXP = /[0-9a-zA-Z\u00C0-\u1FFF\u2C00-\uD7FF]+/g;
+
+        return (contentBlock, callback) => {
+            const text = contentBlock.getText();
+            let matchArr, start, regex = WORD_REGEXP;
+
+            while ((matchArr = regex.exec(text)) !== null) {
+                start = matchArr.index;
+                if (!isCorrectWord(matchArr[0])) {
+                    callback(start, start + matchArr[0].length);
+                }
+            }
+        };
+    }
+
+    /**
+      * @ngdoc method
+      * @name DraftSpellchecker#spellcheckSpan
+      * @return {Object} return the component used to highlight errors
+      * @description Component to highlight errors
+      */
+    spellcheckSpan(props) {
+        const style = styleMap.ERROR;
+        const onContextMenu = (e) => props.showContextMenu(e, props.children[0]);
+
+        return <span style={style} onContextMenu={onContextMenu}>{props.children}</span>;
+    }
+
+    /**
+      * @ngdoc method
+      * @name DraftSpellchecker#showContextMenu
+      * @param {Object} e on context menu event
+      * @param {Object} the component that generates the oncontext menu
+      * @description Show the contextual menu, init the suggestions list
+      */
+    showContextMenu(e, element) {
+        const left = e.clientX - 20;
```

Comment:
Optionally, you could consider using the (undocumented) method `getVisibleSelectionRect` from `draft-js`.

Take a look at the [`LinkPopover.jsx`](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/core/editor3/components/toolbar/LinkPopover.jsx#L118-L127) file, it might help.